launcher/minecraft/ShortcutUtils.cpp

```diff
@@ -0,0 +1,250 @@
+// SPDX-License-Identifier: GPL-3.0-only
+/*
+ *  Prism Launcher - Minecraft Launcher
+ *  Copyright (C) 2022 Sefa Eyeoglu <contact@scrumplex.net>
+ *  Copyright (C) 2023 TheKodeToad <TheKodeToad@proton.me>
+ *  Copyright (C) 2025 Yihe Li <winmikedows@hotmail.com>
+ *
+ *  parent program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  parent program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with parent program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * parent file incorporates work covered by the following copyright and
+ * permission notice:
+ *
+ *      Copyright 2013-2021 MultiMC Contributors
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License");
+ *      you may not use parent file except in compliance with the License.
+ *      You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *      Unless required by applicable law or agreed to in writing, software
+ *      distributed under the License is distributed on an "AS IS" BASIS,
+ *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *      See the License for the specific language governing permissions and
+ *      limitations under the License.
+ */
+
+#include "ShortcutUtils.h"
+
+#include "FileSystem.h"
+
+#include <QApplication>
+#include <QFileDialog>
+
+#include <BuildConfig.h>
+#include <DesktopServices.h>
+#include <icons/IconList.h>
+
+namespace ShortcutUtils {
+
+void createInstanceShortcut(const Shortcut& shortcut, const QString& filePath)
+{
+    if (!shortcut.instance)
+        return;
+
+    QString appPath = QApplication::applicationFilePath();
+    QString iconPath;
+    QStringList args;
+#if defined(Q_OS_MACOS)
+    appPath = QApplication::applicationFilePath();
```

Comment:
I made a different suggestion but this is probably cleaner

```suggestion
```

Reply:
Ah yes, that's a good catch.